pkg/reconciler/v1alpha1/pipelinerun/resources/conditionresolution.go

```diff
@@ -95,24 +99,65 @@ func (rcc *ResolvedConditionCheck) ConditionToTaskSpec() *v1alpha1.TaskSpec {
 		Steps: []corev1.Container{rcc.Condition.Spec.Check},
 	}
 
-	if len(rcc.Condition.Spec.Params) > 0 {
-		t.Inputs = &v1alpha1.Inputs{
-			Params: rcc.Condition.Spec.Params,
-		}
-		// convert param strings of type ${params.x} to ${inputs.params.x}
-		// in order to apply taskrun substitution
-		convertParamTemplates(&t.Steps[0], rcc.Condition.Spec.Params)
+	t.Inputs = &v1alpha1.Inputs{
+		Params:    rcc.Condition.Spec.Params,
+		Resources: toTaskResouces(rcc.Condition.Spec.Resources),
+	}
+
+	// convert param strings of type ${params.x} to ${inputs.params.x}
+	convertParamTemplateStrings(&t.Steps[0], rcc.Condition.Spec.Params, rcc.Condition.Spec.Resources)
+	// convert resource strings of type ${resources.name.key} to ${inputs.resources.name.key}
+	err := convertResourceTemplateStrings(&t.Steps[0], rcc.ResolvedResources, rcc.Condition.Spec.Resources)
+
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to replace resource template strings %w", err)
+	}
+
+	return t, nil
+}
+
+func toTaskResouces(resources []v1alpha1.ConditionResource) []v1alpha1.TaskResource {
+	var tr []v1alpha1.TaskResource
+	for _, r := range resources {
+		tr = append(tr, v1alpha1.TaskResource{
+			Name:       r.Name,
+			Type:       r.Type,
+			TargetPath: r.TargetPath,
+		})
 	}
-	return t
+	return tr
 }
 
-// Replaces all instances of ${params.x} in the container to ${inputs.params.x} for each param name
-func convertParamTemplates(container *corev1.Container, params []v1alpha1.ParamSpec) {
+// Prepends inputs. to all param strings so that they can be replaced by
+// taskrun variable substitution e.g. ${params.x} to ${inputs.params.x}
+func convertParamTemplateStrings(container *corev1.Container, params []v1alpha1.ParamSpec, resources []v1alpha1.ConditionResource) {
 	replacements := make(map[string]string)
 	for _, p := range params {
 		replacements[fmt.Sprintf("params.%s", p.Name)] = fmt.Sprintf("${inputs.params.%s}", p.Name)
-		templating.ApplyContainerReplacements(container, replacements, map[string][]string{})
 	}
+
+	templating.ApplyContainerReplacements(container, replacements, map[string][]string{})
+}
+
+// Prepends inputs. to all resource template strings so that they can be replaced by
+// taskrun variable substitution e.g. ${resources.name.key} to ${inputs.resources.name.key}
+func convertResourceTemplateStrings(container *corev1.Container,
+	resolvedResources map[string]*v1alpha1.PipelineResource,
+	conditionResources []v1alpha1.ConditionResource) error {
```

Comment:
this indentation seems kinda non - standard, makes it a bit hard to read - i think it's okay to just have a really long line for the function signature imo